api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java

```diff
@@ -836,4 +855,43 @@ public CohortDefinition getNumberOfPatientsWithAtLeastOneClinicalAppointmentDuri
     cd.setCompositionString("F1 AND NOT Fx3");
     return cd;
   }
+
+  /**
+   * Get number of patients who have a consultation on the same date as that of pre art
+   *
+   * @return CohortDefinition
+   */
+  private CohortDefinition
+      getPatientsWithFirstClinicalConsultationOnTheSameDateAsPreArtStartDate() {
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.setName(
+        "Get number of patients who have a consultation on the same date as that of pre art");
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.setQuery(
+        ResumoMensalQueries.getPatientsWithFirstClinicalConsultationOnTheSameDateAsPreArtStartDate(
+            hivMetadata.getMasterCardEncounterType().getEncounterTypeId(),
+            hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId(),
+            hivMetadata.getPreArtStartDate().getConceptId()));
+    return cd;
+  }
+
+  /**
+   * B12-B7A: Number of active patients in ART by end of previous/current month
+   *
+   * @retrun CohortDefinition
+   */
+  public CohortDefinition getNumberOfPatientsWhoAbandonedArtDuringPreviousMonthForB127A() {
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.setName("Number of patients who Abandoned the ART during the current month");
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.setQuery(
+        ResumoMensalQueries.getNumberOfPatientsWhoAbandonedArtDuringPreviousMonthB127A(
+            hivMetadata.getARVPharmaciaEncounterType().getEncounterTypeId(),
+            hivMetadata.getMasterCardEncounterType().getEncounterTypeId(),
```

Comment:
@ningosi it should be `hivMetadata.getMasterCardDrugPickupEncounterType().getEncounterTypeId()`